en/docs/learn/design-api/advanced-topics/adding-an-api-state-change-workflow.md

```diff
@@ -62,8 +62,10 @@ First, enable the API state change workflow for **Approval Workflow Executor**.
 
 The elements of the above configuration are explained below.
 
-| Element name                                       | Description                                                                                                                                               |
-|----------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------|                                                                f                                                                  |
-| `stateList`| Mandatory          | This is a comma separated list of the current state and intended action. For example, Created:Publish,Published:Block 
-
-                                                                |
+| Element Name               | Mandatory/Optional | Description                                                                                                                                                                                         |
+|----------------------------|--------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `processDefinitionKey`     | Mandatory          | BPMN process definition ID.  BPMN process provided with AM as default has `APIStateChangeApprovalProcess` as the ID                                                                                 |
+| `stateList`                | Mandatory          | This is a comma separated list of the current state and intended action. For example, Created:Publish,Published:Block                                                                               |
+| `serviceEndpoint`          | Optional           | The URL of the BPMN process engine. This overrides the global `<ServerUrl>` value from the `api-manager.xml` file. This can be used to connect a separate workflow engine for a tenant.             |
```

Comment:
What we have currently is the inbuilt Workflow process. This document explains that. In this case, these configurations do not matter right?

Reply:
We need these configurations to enable WS workflow via carbon console as per documentation.